test/test_callback.py

```diff
@@ -0,0 +1,31 @@
+import threading
+
+from absl.testing import absltest
+import torch
+import torch_xla
+from torch_xla.experimental import callback
+
+
+class TestExperimentalCallback(absltest.TestCase):
+
+  @staticmethod
+  @torch_xla.compile
+  def executable():
+    a, b = torch.randn((100, 100), device=torch_xla.device()), torch.randn(
+        (100, 100), device=torch_xla.device())
+    return a @ b
+
+  def test_callback(self):
+    event = threading.Event()
+    c = self.executable()
+
+    def cb(tensor):
+      self.assertIs(c, tensor)
```

Comment:
this doesn't really check if the computation has finished. Can you check the output of https://github.com/pytorch/xla/blob/master/torch_xla/csrc/init_python_bindings.cpp#L1214-L1217 instead? It will output whether the PJRT buffer is a placeholder or a real value.

Reply:
Fair enough. I just tried this, but I don't think the debug string is really testing that the data is ready either. I see:

```
# before
XLATensor {
TensorID: 5
Device: CPU:0
XLA Shape: f32[100,100]
ShardingSpec: None
IR: None
XLAData: 
  Data Device: CPU:0
  Data Shape: f32[100,100]
  Data Handle: None
Tensor on host: None
}
```

vs

```
# after
XLATensor {
TensorID: 5
Device: CPU:0
XLA Shape: f32[100,100]
ShardingSpec: None
IR: None
XLAData: 
  Data Device: CPU:0
  Data Shape: f32[100,100]
  Data Handle: 139615763718176
Tensor on host: None
}
```

That only tells us that a `PjRtBuffer` has been assigned and not that the buffer is ready. Do you think it's worth adding the buffer state to the debug string, or is it worth adding a new private API like `_is_buffer_ready`?

Reply:
oh right... I forgot that we will have a handle but that handle won't be ready. I think either approach will be fine and we can do that in a follow up pr.